common/src/main/java/com/cinemamod/mcef/MCEF.java

```diff
@@ -107,14 +107,30 @@ private static void assertInitialized() {
 
     public static String getJavaCefCommit() throws IOException {
         // Try to get from resources (if loading from a jar)
-        InputStream inputStream = MCEF.class.getClassLoader().getResourceAsStream("META-INF/MANIFEST.MF");
-        Properties properties = new Properties();
-        try {
-            properties.load(inputStream);
-            if (properties.containsKey("java-cef-commit")) {
-                return properties.getProperty("java-cef-commit");
+
+        Enumeration<URL> resources = MCEF.class.getClassLoader().getResources("META-INF/MANIFEST.MF");
+        Map<String, String> commits = new HashMap<>(1);
+        resources.asIterator().forEachRemaining(resource -> {
+            Properties properties = new Properties();
+            try {
+                properties.load(resource.openStream());
+                if (properties.containsKey("java-cef-commit")) {
+                    commits.put(resource.getFile() , properties.getProperty("java-cef-commit"));
+                }
+            } catch (IOException e) {
+                System.err.println("Failed to get hash");
+                System.err.println(e);
+                e.printStackTrace();
             }
-        } catch (IOException ignored) {
+        });
+
+        if(commits.size() == 1){
+            return commits.get(commits.keySet().stream().toList().get(0));
+        } else if (commits.size() >= 2){
+            System.out.println("multiple values found");
+            commits.forEach((v1, v2) ->{
+                System.out.println(v1 + "  :  " + v2);
+            });
         }
```

Comment:
Can you explain why this is needed? I don't understand why the other code fails or what was the need to do all of this. It's way more complicated.